static/js/services/videowaiter.js

```diff
@@ -46,13 +47,17 @@ define(["underscore"], function(_) {
 				if (videoTracks.length > 0 && this.count >= 10) {
 					var videoTrack = videoTracks[0];
 					if (videoTrack.enabled === true && videoTrack.muted === true) {
+						videoTrack.onunmute = function() {
+							videoTrack.onunmute = undefined;
+							recheck();
```

Comment:
Put this into a setTimeout, so the actual recheck does not run in the event handlers stack. But in generate this is problematic as it does triggers the video waiter a second time. This is not the intention - it should only be triggered once.  I suggest you refactor this, maybe to include a third callback which gets triggered later and only trigger the relevant actions (eg. late_cb parameter).